pyzo/codeeditor/extensions/behaviour.py

```diff
@@ -441,3 +442,90 @@ def _paste(self, keepSelection):
             cursor.setKeepPositionOnInsert(False)
             self.setTextCursor(cursor)
 
+
+class AutoCloseQuotesAndBrackets(object):
+    """
+    Automatic insertion of quotes, parenthesis, braces and brackets
+
+    """
+    
+    def keyPressEvent(self, event):
+        
+        quotesKeys = [Qt.Key_Apostrophe, Qt.Key_QuoteDbl]
+        openBrackets = [Qt.Key_BraceLeft,Qt.Key_BracketLeft, Qt.Key_ParenLeft]
+        closeBrackets = [Qt.Key_BraceRight, Qt.Key_BracketRight, Qt.Key_ParenRight]
+
+        cursor = self.textCursor()
+        
+        # opening brackets
+        if event.key() in openBrackets and pyzo.config.settings.autoClose_Brackets:
+            idx = openBrackets.index(event.key())
+            insert_txt = "{}{}".format(chr(openBrackets[idx]), chr(closeBrackets[idx]))
+            cursor.insertText(insert_txt)
+            self._moveCursorLeft(1)
+        
+        # closing brackets
+        elif event.key() in closeBrackets and pyzo.config.settings.autoClose_Brackets:
+            idx = closeBrackets.index(event.key())
+            next_character = self._getNextCharacter(cursor)
+            if next_character:
+                # skip
+                if ord(next_character) in closeBrackets:
```

Comment:
Should this be `if ord(next_character) in event.key():` or similar?

As it is now, If I am in from of *any* closing bracket, typing any closing bracket will skip over it; it should only work when the match is exact.